WordPress/Classes/Utility/BackgroundTasks/WeeklyRoundupBackgroundTask.swift

```diff
@@ -7,7 +7,8 @@ private class WeeklyRoundupDataProvider {
 
     // MARK: - Definitions
 
-    typealias SiteStats = [Blog: StatsSummaryData]
+    typealias BlogManagedObjectID = NSManagedObjectID
+    typealias SiteStats = [BlogManagedObjectID: StatsSummaryData]
```

Comment:
An instance of `SiteStats` type was used in a queue different than the Blog's context queue. Replaced key type `Blog` with `NSManagedObjectID` for better thread-safety. And users of this instance can fetch the blog instances like this:

```swift
let blog = try? context.existingObject(id) as? Blog
```